frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx

```diff
@@ -222,29 +225,100 @@ export const DecisionForm: FC<props> = ({
-      if (response === "success") {
-        //-- update the assignment of the complaint to the selected officer
-        const { assignedTo } = data;
-        const result = await dispatch(assignComplaintToOfficer(leadIdentifier, assignedTo));
+    if (isValid()) {
+      dispatch(upsertDecisionOutcome(identifier, data)).then(async (response) => {
+        if (response === "success") {
+          //-- update the assignment of the complaint to the selected officer
+          const { assignedTo } = data;
+          if (assignedTo) {
+            const result = await dispatch(assignComplaintToOfficer(leadIdentifier, assignedTo));
 
-        if (result) {
-          //-- update the complaint
-          dispatch(getComplaintById(leadIdentifier, COMPLAINT_TYPES.ERS));
-        } else {
-          ToggleError("Error, unable to to assign officer to complaint");
-        }
+            if (result) {
+              //-- update the complaint
+              dispatch(getComplaintById(leadIdentifier, COMPLAINT_TYPES.ERS));
+            } else {
+              ToggleError("Error, unable to to assign officer to complaint");
+            }
+          }
 
-        dispatch(getCaseFile(leadIdentifier));
+          dispatch(getCaseFile(leadIdentifier));
 
-        if (id !== undefined) {
-          toggleEdit(false);
+          if (id !== undefined) {
+            toggleEdit(false);
+          }
         }
+      });
+    }
+  };
+
+  const isValid = (): boolean => {
+    let _isValid = true;
+
+    if (!data.schedule) {
+      setScheduleErrorMessage("Required");
+    }
+
+    if (!data.sector) {
+      setSectorErrorMessage("Required");
+    }
+
+    if (!data.discharge) {
+      setDischargeErrorMessage("Required");
+    }
+
+    if (!data.schedule || !data.sector || !data.discharge) {
+      _isValid = false;
+    }
+
+    if (data.actionTaken === CASE_ACTION_CODE.FWDLEADAGN && !data.leadAgency) {
+      setLeadAgencyErrorMessage("Required");
+      _isValid = false;
+    }
+
+    if (data.actionTaken === CASE_ACTION_CODE.RESPREC && !data.inspectionNumber) {
+      setInspectionNumberErrorMessage("Required");
+      _isValid = false;
+    }
+
+    if (data.actionTaken && (!data.actionTakenDate || !data.assignedTo)) {
+      if (!data.actionTakenDate) {
+        setDateActionTakenErrorMessage("Date required when action taken selected");
+        _isValid = false;
       }
-    });
+      if (!data.assignedTo) {
+        setAssignedToErrorMessage("Assigned to required when action taken selected");
+        _isValid = false;
+      }
+    }
+
+    if (data.assignedTo && (!data.actionTaken || !data.actionTakenDate)) {
+      if (!data.actionTakenDate) {
+        setDateActionTakenErrorMessage("Date required when action taken selected");
+        _isValid = false;
+      }
+
+      if (!data.actionTaken) {
+        setActionTakenErrorMessage("Action taken required when assigned to is selected");
+        _isValid = false;
+      }
+    }
```

Comment:
If this was moved into it's own function (e.g. validateActionTaken), could cognitive complexity be reduced?

Reply:
I think this should be easier for sonar cloud though I really don't like it